Assignment5/GenericsHomwork.Tests/NodeTests.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using GenericsHomework;
+using System;
+
+namespace GenericsHomwork.Tests
+{
+    [TestClass]
+    public class NodeTests
+    {
+        [TestMethod]
+        public void Constructor_ValidInput_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number);
+            Assert.IsNotNull(testNode);
+        }
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void Constructor_NullInput_Success()
+        {
+            int? number = null;
+            Node<int?> testNode = new Node<int?>(number);
+        }
+        [TestMethod]
+        public void Constructor_NullNext_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number, null);
+            Assert.IsNotNull(testNode);
+        }
+        [TestMethod]
+        public void Insert_ValidInput_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number);
+            testNode.Insert(number);
+            Assert.IsNotNull(testNode);

```

Comment:
I would expect this test to also tests against the `testProperty.Next`. I would also insert a different value than what was passed in the constructor; just to make it easier to know which node we are looking at.

Something more like (assumes Value property getter is made public):
```suggestion
            Node<int> testNode = new(4);
            testNode.Insert(42);
            Assert.AreEqual(4, testNode.Value);
            Assert.AreEqual(42, testNode.Next.Value);
```